packages/rrweb/src/record/observers/canvas/canvas-manager.ts

```diff
@@ -305,28 +312,25 @@ export class CanvasManager implements CanvasManagerInterface {
     this.pendingCanvasMutations.get(target)!.push(mutation);
   };
 
-  private initCanvasFPSObserver(
-    fps: number,
-    blockClass: blockClass,
-    blockSelector: string | null,
-    unblockSelector: string | null,
-    maxCanvasSize: [number, number] | null | undefined,
-    options: {
-      dataURLOptions: DataURLOptions;
-    },
-  ) {
-    const rafId = this.takeSnapshot(
-      false,
-      fps,
-      blockClass,
-      blockSelector,
-      unblockSelector,
-      maxCanvasSize,
-      options.dataURLOptions,
-    );
+  private initCanvasFPSObserver() {
+    let rafId: number;
+
+    if (!this.windows.length && !this.shadowDoms.size) {
+      // If these are empty, then we won't be able to find any canvases to snapshot, so nothing to do here.
+      return;
+    }
+
+    const rafCallback = (timestamp: DOMHighResTimeStamp) => {
+      this.takeSnapshot(timestamp, false);
+      rafId = onRequestAnimationFrame(rafCallback);
+    };
+
+    rafId = onRequestAnimationFrame(rafCallback);
 
     this.restoreHandlers.push(() => {
-      cancelAnimationFrame(rafId);
```

Comment:
This didn't work as expected since `takeSnapshot()` loops on itself and this `rafId` is only for the first call.